File: examples/sakila/python/02_gsi_query.py
from __future__ import annotations

import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))

from common import load_db


def main() -> None:
    db = load_db()
    query = {
        "customers": {
            "index": "by_email",
            "where": {"email": "CLARA.SHAW@sakilacustomer.org"},
        }
    }
    result = db.execute(query)
    print(json.dumps(result, indent=2))


if __name__ == "__main__":
    main()

File: examples/sakila/rust/src/bin/02_gsi_query.rs
use serde_json::json;
use virtuus_sakila_examples::load_db;

fn main() {
    let mut db = load_db();
    let query = json!({
        "customers": {
            "index": "by_email",
            "where": {"email": "CLARA.SHAW@sakilacustomer.org"}
        }
    });
    let result = db.execute(&query);
    println!("{}", serde_json::to_string_pretty(&result).unwrap());
}
